synapse/config/database.py

```diff
@@ -127,27 +128,41 @@ def generate_config_section(self, data_dir_path, database_conf, **kwargs):
         )
 
     def read_arguments(self, args):
-        self.set_databasepath(args.database_path)
+        """
+        Cases for the cli input:
+          - If there are multiple databases and a path raise an error.
+          - If the database set in the config file is sqlite then
+            overwrite with the command line argument.
+          - If no databases are configured and no path available raise.
+          - No databases and only path available ==> sqlite3 db.
+        """
+        databases = getattr(self, "databases")
+        if databases:
+            # multiple databases from the config file
+            if len(databases) > 1 and args.database_path:
+                raise ConfigError("More than one database exists")
+
+            # single database from the config file
+            if self.databases[0].name == "sqlite3" and args.database_path:
+                self.set_databasepath(args.database_path)
+            else:
+                logger.warn("Ignoring 'database_path' for non-sqlite3 database")
+
+        else:
+            # no databases from the config file
+            if args.database_path is None:
+                raise ConfigError("No database config provided")
+            else:
+                database_config = {"name": "sqlite3", "args": {}}
+                self.databases = [DatabaseConnectionConfig("master", database_config)]
+                self.set_databasepath(args.database_path)
```

Comment:
I think the flow in this function would be easier to follow if you dealt with the case of `args.database_path is None` first:

```
if args.database_path is None:
    if not databases:
        raise ConfigError...
   return
```

generally I prefer early `returns` to nested `if/else` statements: they functions easier to follow.